### PR TITLE
new repo tracks user-bookmark clicks

### PR DESCRIPTION
**Pull request by yingjieMiao**

### CODE REVIEW

**Comment by yingjieMiao** [comment]
@leogrim 
@stkem 
@ymatsuda

File: kifi-backend/modules/shoebox/app/com/keepit/model/UserBookmarkClicksRepo.scala [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import com.keepit.common.db.slick.DBSession._
+import com.keepit.common.db._
+import com.keepit.common.time._
+import com.google.inject.{Singleton, ImplementedBy, Inject}
+
+
+@ImplementedBy(classOf[UserBookmarkClicksRepoImpl])
+trait UserBookmarkClicksRepo extends Repo[UserBookmarkClicks]{
+  def getByUserUri(userId: Id[User], uriId: Id[NormalizedURI])(implicit session: RSession): Option[UserBookmarkClicks]
+  def increamentCounts(userId: Id[User], uriId: Id[NormalizedURI], isSelf: Boolean)(implicit session: RWSession): UserBookmarkClicks
```
Comment (gmethvin) [inline kifi-backend/modules/shoebox/app/com/keepit/model/UserBookmarkClicksRepo.scala]: `incrementCounts`

Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/model/UserBookmarkClicksRepo.scala]: Got it.

File: kifi-backend/modules/shoebox/app/com/keepit/model/UserBookmarkClicksRepo.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    def otherClicks = column[Int]("other_clicks", O.NotNull)
+    def * = id.? ~ createdAt ~ updatedAt ~ userId ~ uriId ~ selfClicks ~ otherClicks <> (UserBookmarkClicks.apply _, UserBookmarkClicks.unapply _)
+  }
+
+  def getByUserUri(userId: Id[User], uriId: Id[NormalizedURI])(implicit session: RSession): Option[UserBookmarkClicks] = {
+    (for( r<- table if (r.userId === userId && r.uriId === uriId) ) yield r).firstOption
+  }
+
+  def increaseCounts(userId: Id[User], uriId: Id[NormalizedURI], isSelf: Boolean)(implicit session: RWSession): UserBookmarkClicks = {
+    val old = getByUserUri(userId, uriId)
```
Comment (gmethvin) [inline kifi-backend/modules/shoebox/app/com/keepit/model/UserBookmarkClicksRepo.scala]: Might be clearer to do something like:

``` scala
val r = getByUserUri(userId, uriId).getOrElse(
  UserBookmarkClicks(userId = userId, uriId = uriId, selfClicks = 0, otherClicks = 0))
save(if (isSelf) r.copy(selfClicks = r.selfClicks + 1) else r.copy(otherClicks = r.otherClicks + 1))
```

It's one more copy when no record exists, but that is the edge case, right?

Comment (yingjieMiao) [inline kifi-backend/modules/shoebox/app/com/keepit/model/UserBookmarkClicksRepo.scala]: agreed

**Pull request merged**